core/exceptions/ExceptionStackTraceDisplay.php

```diff
@@ -41,191 +53,185 @@ public function __construct(Exception $exception)
      */
     protected function displayException(Exception $exception)
     {
-        $error_code = '';
-        $trace_details = '';
-        $time = time();
-        $trace = $exception->getTrace();
         // get separate user and developer messages if they exist
         $msg = explode('||', $exception->getMessage());
         $user_msg = $msg[0];
         $dev_msg = isset($msg[1]) ? $msg[1] : $msg[0];
         $msg = WP_DEBUG ? $dev_msg : $user_msg;
+        // process trace info
+        $trace_details = $this->traceDetails($exception);
+        $code          = $exception->getCode() ?: $this->error_code;
+        // add helpful developer messages if debugging is on
+        // or generic non-identifying messages for non-privileged users
+        $error_message = WP_DEBUG
+            ? $this->developerError($exception, $msg, $code, $trace_details)
+            : $this->genericError($msg, $code);
         // start gathering output
-        $output = $this->exceptionStyles();
-        $output .= '
-<div id="ee-error-message" class="error">';
-        if (! WP_DEBUG) {
-            $output .= '
-	<p>';
+        $output = '
+<div id="ee-error-message" class="error">
+    ' . $error_message . '
+</div>';
+        $styles = $this->exceptionStyles();
+        $scripts = $this->printScripts(true);
+        if (defined('DOING_AJAX')) {
```

Comment:
Should we also check for the value of this constant ?

Reply:
it is only ever set to `true` by WP core, so unless a third party is setting that constant to `false`,  it's value is kinda moot... 
ie: it's either `true` or undefined

Reply:
Right, ok.